### DISCUSSION TITLE
building filtering / sorting functionality like ransack for ruby

### DISCUSSION THREAD

**Discussion by matthewhausman** [strawberry-graphql/strawberry#3117] [General]
I'm curious how anyone here has built a sophisticated application without the ability to do sophisticated sorting / filters on the server side...

https://activerecord-hackery.github.io/ransack/getting-started/search-matches/

I have code that does this for searching sorting for strawberry but it sucks, how can we make this happen?

**Comment by matthewhausman** [comment]
am I supposed to use this? or is there additional lock in that may be undesirable?

https://github.com/strawberry-graphql/strawberry-graphql-django/blob/main/docs/guide/filters.md

**Reply by patrick91** [reply]
what framework are you using? django? if so, yes 😊

**Reply by matthewhausman** [reply]
None… just an orm for mongo db

**Reply by patrick91** [reply]
so the filters you linked are only for django :)

We don't support mongo as an extension, so you'd need to write your own filters and logic, you mentioned that you have some code already, but you don't like it, why?
